service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java

```diff
@@ -488,7 +488,7 @@ private void setupBamCompleteHandler(Gateway gateway,
             @Override
             public void onMessage(Message msg, Object message) {
                 try {
-                    String bamCompleteJson = mapper.readValue(
+                    String bamCompleteJson = mapper.convertValue(
                         new String(msg.getData(), StandardCharsets.UTF_8),
                         String.class);
```

Comment:
I assume `new String(...)` already returns an object instance of type String. Could you help me understand the purpose of `convertValue()` here?

Reply:
the method call was changed from `readValue(....)` to `convertValue(...)` which behave slightly differently when reading directly from a string or a generic (java) object.